### PR TITLE
feat(blocks): Add jina fact checker block

### PR DESCRIPTION
**Pull request by Bentlybro**
### Background

This adds the new Jina fact checker api as a block
https://jina.ai/news/fact-checking-with-new-grounding-api-in-jina-reader/

video to show it working with the new auth system!

https://github.com/user-attachments/assets/9636a4b6-a207-4f00-b81e-11cca9f901b8

### CODE REVIEW

File: autogpt_platform/backend/backend/blocks/search.py [hunk 1]
```diff
@@ -163,3 +163,59 @@ def run(self, input_data: Input, **kwargs) -> BlockOutput:
             yield "condition", weather_data["weather"][0]["description"]
         else:
             raise RuntimeError(f"Expected keys not found in response: {weather_data}")
+
+
+class FactCheckerBlock(Block, GetRequest):
+    class Input(BlockSchema):
+        statement: str
+        api_key: BlockSecret = SecretField(key="jina_api_key")
```
Comment (aarushik93) [inline autogpt_platform/backend/backend/blocks/search.py]: We have the new credentials manager now - let's use that

File: autogpt_platform/backend/.env.example [hunk 2]
```diff
@@ -95,6 +95,9 @@ REPLICATE_API_KEY=
 # Ideogram
 IDEOGRAM_API_KEY=
 
+# Jina
```
Comment (ntindle) [inline autogpt_platform/backend/.env.example]: Does this work if you set it and don't on the credential?

**Review by ntindle — COMMENTED** [review]
I've made some changes in https://github.com/Significant-Gravitas/AutoGPT/pull/8403 that will likely impact this

validate that the API key works via .env as I wouldn't expect that to work with credits and what not based on current system design

File: autogpt_platform/backend/.env.example (hunk 2 above)
Comment (Bentlybro) [reply autogpt_platform/backend/.env.example]: just tested and no it does not work if i set the key in the env and not the new creds setup so i will remove this from the PR

It mentions that the key is required

image

Comment (ntindle) [reply autogpt_platform/backend/.env.example]: I’m okay with this for now if you are

File: autogpt_platform/backend/backend/blocks/search.py [hunk 3]
```diff
@@ -180,3 +190,82 @@ def run(self, input_data: Input, **kwargs) -> BlockOutput:
             yield "condition", weather_data["weather"][0]["description"]
         else:
             raise RuntimeError(f"Expected keys not found in response: {weather_data}")
+
+
+class FactCheckerBlock(Block, GetRequest):
```
Comment (ntindle) [inline autogpt_platform/backend/backend/blocks/search.py]: move to /jina folder and reuse the _auth.py there

**Review by ntindle — CHANGES_REQUESTED** [review]

File: autogpt_platform/backend/backend/blocks/search.py (hunk 3 above)
Comment (Bentlybro) [reply autogpt_platform/backend/backend/blocks/search.py]: will do :)

File: autogpt_platform/backend/.env.example (hunk 2 above)
Comment (ntindle) [reply autogpt_platform/backend/.env.example]: fixed in my pr #8403 for reference on how it'll need to go in

**Comment by ntindle** [comment]
@Bentlybro whats going on with this pr?

**Comment by Bentlybro** [comment]
@ntindle ill get this sorted today

File: autogpt_platform/backend/backend/blocks/search.py (hunk 3 above)
Comment (Bentlybro) [reply autogpt_platform/backend/backend/blocks/search.py]: Done!

File: autogpt_platform/backend/backend/blocks/jina/fact_checker.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        factuality: float = SchemaField(
+            description="The factuality score of the statement"
+        )
+        result: bool = SchemaField(description="The result of the factuality check")
+        reason: str = SchemaField(description="The reason for the factuality result")
+        error: str = SchemaField(description="Error message if the check fails")
+
+    def __init__(self):
+        super().__init__(
+            id="a1b2c3d4-5e6f-7g8h-9i0j-k1l2m3n4o5p6",
```
Comment (ntindle) [inline autogpt_platform/backend/backend/blocks/jina/fact_checker.py]: gen a new ID

**Review by ntindle — CHANGES_REQUESTED** [review]
Generate a new ID but after that it should be good

File: autogpt_platform/backend/backend/blocks/jina/fact_checker.py (hunk 4 above)
Comment (Bentlybro) [reply autogpt_platform/backend/backend/blocks/jina/fact_checker.py]: oops good spot thanks, fixed :)

**Review by aarushik93 — APPROVED** [review]

**Pull request merged**